common/persistence/sql/sqlExecutionManager.go

```diff
@@ -99,18 +116,16 @@ func (m *sqlExecutionManager) createWorkflowExecutionTx(
 
 	var err error
 	var row *sqldb.CurrentExecutionsRow
-	workflowID := *request.Execution.WorkflowId
-	domainID := sqldb.MustParseUUID(request.DomainID)
 	if row, err = lockCurrentExecutionIfExists(tx, m.shardID, domainID, workflowID); err != nil {
 		return nil, err
 	}
 
-	lastWriteVersion := common.EmptyVersion
-
 	if row != nil {
 		switch request.CreateWorkflowMode {
 		case p.CreateWorkflowModeBrandNew:
-			if request.ReplicationState != nil {
+			// TODO why there is a check here?
```

Comment:
probably for backwards compatibility reasons when global domain support was initially added and lastWriteVersion was not a thing before. anyways, let's chat and resolve this. if you add a TODO here, its likely to remain there for a very long time without resolution